flink-ai-flow/ai_flow/store/sqlalchemy_store.py

```diff
@@ -749,6 +750,150 @@ def delete_model_version_relation_by_version(self, version, model_id) -> Status:
+
+        with self.ManagedSessionMaker() as session:
+            project = self.get_project_by_name(project_name)
+            if not project:
+                raise AIFlowException("The project name you specific doesn't exists, project: \"{}\""
+                                      .format(project_name))
+            workflow_result = session.query(SqlWorkflow).filter(SqlWorkflow.project_id == project.uuid,
+                                                                SqlWorkflow.is_deleted.is_(False)).limit(
+                page_size).offset(offset).all()
+            if len(workflow_result) == 0:
+                return None
+            workflow_list = []
+            for workflow in workflow_result:
+                workflow_list.append(ResultToMeta.result_to_workflow_meta(workflow))
+            return workflow_list
+
+    def delete_workflow_by_name(self, project_name, workflow_name) -> Status:
+        """
+        Delete the workflow by project and workflow name
+
+        :param project_name: the name of project which contains the workflow
+        :param workflow_name: the workflow name
+        """
+        with self.ManagedSessionMaker() as session:
+            try:
+                project = self.get_project_by_name(project_name)
+                if not project:
+                    raise AIFlowException("The project name you specific doesn't exists, project: \"{}\""
+                                          .format(project_name))
+                workflow = session.query(SqlWorkflow).filter(SqlWorkflow.name == workflow_name,
+                                                             SqlWorkflow.project_id == project.uuid,
+                                                             SqlWorkflow.is_deleted.is_(False)).scalar()
+                if workflow is None:
+                    return Status.ERROR
+                deleted_workflow_counts = session.query(SqlWorkflow).filter(
+                    SqlWorkflow.project_id == project.uuid,
+                    SqlWorkflow.name.like(deleted_character + workflow_name + deleted_character + '%'),
+                    SqlWorkflow.is_deleted.is_(True)).count()
+                workflow.is_deleted = True
+                workflow.name = deleted_character + workflow.name + deleted_character + str(deleted_workflow_counts + 1)
+                session.flush()
+                return Status.OK
+            except sqlalchemy.exc.IntegrityError as e:
+                raise AIFlowException(str(e))
+
+    def delete_workflow_by_id(self, workflow_id) -> Status:
+        """
+        Delete the workflow by specific id
+
+        :param workflow_id: the uuid of workflow
+        """
+        workflow = self.get_workflow_by_id(workflow_id)
+        project = self.get_project_by_id(workflow.project_id)
+        if workflow is None or project is None:
+            return Status.ERROR
+        return self.delete_workflow_by_name(project_name=project.name, workflow_name=workflow.name)
```

Comment:
Why not delete workflow directly by uuid? IMO, UUID is primary key for the `workflow` table.